conjure-java-core/src/integrationInput/java/com/palantir/product/MultipleOrderedStages.java

```diff
@@ -0,0 +1,252 @@
+
+    private int memoizedHashCode;
+
+    private MultipleOrderedStages(
+            OneField token, String item, Set<SafeLong> items, Map<ResourceIdentifier, String> mappedRids) {
+        validateFields(token, item, items, mappedRids);
+        this.token = token;
+        this.item = item;
+        this.items = Collections.unmodifiableSet(items);
+        this.mappedRids = Collections.unmodifiableMap(mappedRids);
+    }
+
+    @JsonProperty("token")
+    public OneField getToken() {
+        return this.token;
+    }
+
+    @JsonProperty("item")
+    public String getItem() {
+        return this.item;
+    }
+
+    @JsonProperty("items")
+    public Set<SafeLong> getItems() {
+        return this.items;
+    }
+
+    @JsonProperty("mappedRids")
+    public Map<ResourceIdentifier, String> getMappedRids() {
+        return this.mappedRids;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return this == other || (other instanceof MultipleOrderedStages && equalTo((MultipleOrderedStages) other));
+    }
+
+    private boolean equalTo(MultipleOrderedStages other) {
+        return this.token.equals(other.token)
+                && this.item.equals(other.item)
+                && this.items.equals(other.items)
+                && this.mappedRids.equals(other.mappedRids);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = memoizedHashCode;
+        if (result == 0) {
+            result = Objects.hash(this.token, this.item, this.items, this.mappedRids);
+            memoizedHashCode = result;
+        }
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "MultipleOrderedStages{token: " + token + ", item: " + item + ", items: " + items + ", mappedRids: "
+                + mappedRids + '}';
+    }
+
+    private static void validateFields(
+            OneField token, String item, Set<SafeLong> items, Map<ResourceIdentifier, String> mappedRids) {
+        List<String> missingFields = null;
+        missingFields = addFieldIfMissing(missingFields, token, "token");
+        missingFields = addFieldIfMissing(missingFields, item, "item");
+        missingFields = addFieldIfMissing(missingFields, items, "items");
+        missingFields = addFieldIfMissing(missingFields, mappedRids, "mappedRids");
+        if (missingFields != null) {
+            throw new SafeIllegalArgumentException(
+                    "Some required fields have not been set", SafeArg.of("missingFields", missingFields));
+        }
+    }
+
+    private static List<String> addFieldIfMissing(List<String> prev, Object fieldValue, String fieldName) {
+        List<String> missingFields = prev;
+        if (fieldValue == null) {
+            if (missingFields == null) {
+                missingFields = new ArrayList<>(4);
+            }
+            missingFields.add(fieldName);
+        }
+        return missingFields;
+    }
+
+    public static ItemStageBuilder builder() {
```

Comment:
@carterkozak would recommend taking a look at this example test that demonstrates the implementation we discussed

Reply:
Actually, let me adjust it such that methods on `Builder` interface all return `Builder` - That way, the consumers of this won't have to worry about following ordering if they've reached the `Builder` position (which can be only be reached from a `from` (he he)

Reply:
Done in https://github.com/palantir/conjure-java/pull/1243/commits/4c8b0beafff350427e7d47555a6bc6e9bad2b6cb